common/jdbc/src/main/scala/ai/h2o/mojos/deploy/common/jdbc/SqlScorerClient.scala

```diff
@@ -0,0 +1,192 @@
+    logger.info("Recieved request to score sql query")
+    logger.info(
+      String.format(
+        "Query inputs: query - %s, outputTable - %s, idColumn - %s",
+        scoreRequest.getSqlQuery,
+        scoreRequest.getOutputTable,
+        scoreRequest.getIdColumn
+      )
+    )
+    val sqlQuery: String = String.format(
+      "(%s) queryTable",
+      sanitizeInputString(scoreRequest.getSqlQuery)
+    )
+    if (scoreRequest.getIdColumn != null) {
+      logger.info("Generating partitions for query: {}", sqlQuery)
+      generatePartitionMapping(
+        jdbcConfig.dbConnectionString,
+        sqlQuery,
+        sanitizeInputString(scoreRequest.getIdColumn)
+      )
+    }
+    logger.info("Executing query")
+    doQuery(jdbcConfig.dbConnectionString, sqlQuery)
+  }
+
+  private def write(scoreRequest: ScoreRequest, predsDf: DataFrame): Unit = {
+    logger.info("Writing scored data to table")
+    predsDf
+      .write
+      .mode(castSaveModeFromRequest(scoreRequest))
+      .jdbc(
+        url = jdbcConfig.dbConnectionString,
+        table = scoreRequest.getOutputTable,
+        sqlProperties
+      )
+  }
+
+  private def loadMojoPipelineFromFile(): H2OMOJOPipelineModel = {
+    Preconditions.checkArgument(
+      !Strings.isNullOrEmpty(MOJO_PIPELINE_PATH),
+      "Path to mojo pipeline not specified, set the %s property",
+      MOJO_PIPELINE_PATH_PROPERTY
+    )
+    logger.info("Spark: Loading Mojo pipeline from path {}", MOJO_PIPELINE_PATH)
+    var mojoFile: File = new File(MOJO_PIPELINE_PATH)
+    if (!mojoFile.isFile) {
+      val classLoader: ClassLoader = SqlScorerClient.getClass.getClassLoader
+      val resourcePath: URL = classLoader.getResource(MOJO_PIPELINE_PATH)
+      if (resourcePath != null) {
+        mojoFile = new File(resourcePath.getFile)
+      }
+    }
+    if (!mojoFile.isFile) {
+      throw new RuntimeException("Could not load mojo")
+    }
+    try {
+      val pipeline: H2OMOJOPipelineModel = H2OMOJOPipelineModel.createFromMojo(mojoFile.getPath)
+      logger.info("Mojo pipeline successfully loaded ({})", pipeline.uid)
+      pipeline
+    } catch {
+      case e: IOException =>
+        throw new RuntimeException("Unable to load mojo", e)
+      case e: LicenseException =>
+        throw new RuntimeException("License file not found", e)
+    }
+  }
+
+  private def loadJdbcConfigFromFile(): JdbcConfig = {
+    Preconditions.checkArgument(
+      !Strings.isNullOrEmpty(JDBC_CONFIG_FILE),
+      "Path to jdbc config file is not specified, set the %s property",
+      JDBC_CONFIG_FILE_PROPERTY
+    )
+    logger.info("Loading the JDBC Config file from path {}", JDBC_CONFIG_FILE)
+    try {
+      new JdbcConfig(JDBC_CONFIG_FILE)
+    } catch {
+      case e: IOException =>
+        throw new RuntimeException("Unable to load JDBC Config file", e)
+    }
+  }
+
+  private def generatePartitionMapping(url: String, query: String, idColumn: String): Unit = {
+    val tmpDf: DataFrame = doQuery(url, query)
```

Comment:
Just a note: Most of Scala developers don't specify type of variables and private fields. It's definitely a matter of preferences, just mentioning it.

Reply:
hmm yea just habitual from java I'd say... if its not overly terrifying, then I'd lean towards keeping the type defs and private fields, if only because I'm a little obsessive compulsive :smile: